src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useFlakeAggregates/useFlakeAggregates.tsx

```diff
@@ -0,0 +1,120 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import { MeasurementInterval } from 'pages/RepoPage/shared/constants'
+import { RepoNotFoundErrorSchema } from 'services/repo'
+import Api from 'shared/api'
+import { NetworkErrorObject, rejectNetworkError } from 'shared/api/helpers'
+
+const FlakeAggregatesSchema = z.object({
+  owner: z
+    .object({
+      repository: z.discriminatedUnion('__typename', [
+        z.object({
+          __typename: z.literal('Repository'),
+          testAnalytics: z
+            .object({
+              flakeAggregates: z.object({
+                flakeCount: z.number(),
+                flakeCountPercentChange: z.number().nullable(),
+                flakeRate: z.number(),
+                flakeRatePercentChange: z.number().nullable(),
+              }),
+            })
+            .nullable(),
+        }),
+        RepoNotFoundErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `
+  query GetFlakeAggregates(
+    $owner: String!
+    $repo: String!
+    $interval: MeasurementInterval
+  ) {
+    owner(username: $owner) {
+      repository: repository(name: $repo) {
+        __typename
+        ... on Repository {
+            testAnalytics {
+              flakeAggregates(interval: $interval) {
+                flakeCount
+                flakeCountPercentChange
+                flakeRate
+                flakeRatePercentChange
+              }
+          }
+        }
+        ... on NotFoundError {
+          message
```

Comment:
`m`: Wondering if we can add in handle this potential union type `OwnerNotActivatedError`? What do you think about handling the error, with our schema validation, and also in our resolver, until it has been removed from the schema fully?

Reply:
I'm certain we don't need this union type, as it's already handled in the repository-level validations. Duplicating it here would just add dead code